components/mediation-admin/org.wso2.carbon.endpoint/src/main/java/org/wso2/carbon/endpoint/service/EndpointAdmin.java

```diff
@@ -164,9 +164,8 @@ public boolean addEndpoint(String epString) throws EndpointAdminException {
                 assertNameNotEmpty(endpointName);
                 endpointName = endpointName.trim();
                 log.debug("Adding endpoint : " + endpointName + " to the configuration");
-
-                if (getSynapseConfiguration().getLocalRegistry()
-                            .get(endpointName) != null) {
+                Object entry = getSynapseConfiguration().getLocalRegistry().get(endpointName);
+                if (entry != null && entry instanceof AbstractEndpoint) {
```

Comment:
Why do we need to do an instance of check here? Aren't we overidding non-endpoint entries if we do this?

Reply:
Otherwise, it will return the same error messages for all kind of entries. Added this due to adding of temporary entry during the  'getEntryDefinition' in 'synapse'.

Reply:
Let's say we already have a registry artifact name similar to the endpoint location we are creating. If we have this check that artifact will get replaced. Isn't it?